qiskit/transpiler/passes/scheduling/alap.py

```diff
@@ -68,71 +47,89 @@ def run(self, dag):
         bit_indices = {bit: index for index, bit in enumerate(dag.qubits)}
         for node in reversed(list(dag.topological_op_nodes())):
-            # validate node.op.duration
-            if node.op.duration is None:
-                indices = [bit_indices[qarg] for qarg in node.qargs]
-                if dag.has_calibration_for(node):
-                    node.op.duration = dag.calibrations[node.op.name][
-                        (tuple(indices), tuple(float(p) for p in node.op.params))
-                    ].duration
-
-                if node.op.duration is None:
-                    raise TranspilerError(
-                        f"Duration of {node.op.name} on qubits {indices} is not found."
-                    )
-            if isinstance(node.op.duration, ParameterExpression):
-                indices = [bit_indices[qarg] for qarg in node.qargs]
-                raise TranspilerError(
-                    f"Parameterized duration ({node.op.duration}) "
-                    f"of {node.op.name} on qubits {indices} is not bounded."
-                )
-            # choose appropriate clbit available time depending on op
-            clbit_time_available = (
-                clbit_writeable if isinstance(node.op, Measure) else clbit_readable
-            )
-            # correction to change clbit start time to qubit start time
-            delta = 0 if isinstance(node.op, Measure) else node.op.duration
-            # must wait for op.condition_bits as well as node.cargs
-            start_time = max(
-                itertools.chain(
-                    (qubit_time_available[q] for q in node.qargs),
-                    (clbit_time_available[c] - delta for c in node.cargs + node.op.condition_bits),
-                )
-            )
-
-            pad_with_delays(node.qargs, until=start_time, unit=time_unit)
+            op_duration = self._get_node_duration(node, bit_indices, dag)
+
+            # compute t0, t1: instruction interval, note that
+            # t0: start time of instruction
+            # t1: end time of instruction
+
+            # since this is alap scheduling, node is scheduled in reversed topological ordering
+            # and nodes are packed from the very end of the circuit.
+            # the physical meaning of t0 and t1 is flipped here.
+            if isinstance(node.op, Measure):
+                # clbit time is always right (alap) justified
+                t0 = max(idle_before[bit] for bit in node.qargs + node.cargs)
+                t1 = t0 + op_duration
+                #
+                #        |t1 = t0 + duration
+                # Q ░░░░░▒▒▒▒▒▒▒▒▒▒▒
+                # C ░░░░░░░░░▒▒▒▒▒▒▒
+                #            |t0 + (duration - clbit_write_latency)
+                #
+                for clbit in node.cargs:
+                    idle_before[clbit] = t0 + (op_duration - self.clbit_write_latency)
+            elif isinstance(node.op, Gate):
```

Comment:
```suggestion
            elif isinstance(node.op, Gate) or isinstance(node.op, Delay):
```
Can we support conditional delays as well?